pkg/snapstore/abs_snapstore.go

```diff
@@ -72,35 +74,41 @@ func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 		Retry: azblob.RetryOptions{
 			TryTimeout: downloadTimeout,
 		}})
-	u, err := url.Parse(fmt.Sprintf("https://%s.%s", storageAccount, brtypes.AzureBlobStorageHostName))
+
+	absURI := brtypes.AzureBlobStorageGlobalURI
+	// configuration provides an override for Azure's URI
+	if overrideURI != nil {
+		absURI = *overrideURI
+	}
```

Comment:
```suggestion
	absURI := brtypes.AzureBlobStorageGlobalURI
	// configuration provides an override for Azure's URI
	if overrideURI != nil && !config.IsSource {
		absURI = *overrideURI
	}
```

For copy operation, we should not set the endpoint override as the buckets may reside on different regions. Refer the way it's done for gcs [here](https://github.com/gardener/etcd-backup-restore/blob/b311db7f4c9692b4f887319a4fa115fcba5c7e02/pkg/snapstore/gcs_snapstore.go#L65-L70)
cc @shreyas-s-rao wdyt?

Reply:
Great find @anveshreddy18!
I unfortunately forgot checking the implementation of GCS for this.

However, I see one issue with the suggestion.
If a copy operation is to be performed between two buckets in the *same* region which both need an overridden URI for ABS, then this would fail. Should it be etcd-backup-restore's responsibility to disallow copy of contents in buckets, @shreyas-s-rao?

Reply:
@anveshreddy18 
>For copy operation, we should not set the endpoint override as the buckets may reside on different regions.

Can you explain this further? I also had a look at the GCS code along with @renormalize  and the implementation does not prevent any such thing. Also in case we have missed the code that actually does it, i agree with @renormalize  that it is not a responsibility of backup-restore to prevent such copies. Given 2 stores it copies from source to target and that is it.

Reply:
>it is not a responsibility of backup-restore to prevent such copies

@unmarshall  I agree with your latter point. For explaining the first part, what I meant was that the functionality to disable copy between buckets of different region is implemented in the `etcdbr` even though that is not being consumed by druid now. I guess this was not coded in druid after this [PR](https://github.com/gardener/etcd-backup-restore/pull/691) in backup-restore. 

I believe there are both pros and cons to that approach ( gcs way ) as it will prevent the the copy if one region is override region, and other is normal. Because with the current way ( azure way ), to do that kind of copy between different subdomains, we would probably need an additional field in the secret like `destEndpoint`. I was thinking in this point of view. 

Overall, I am fine with not having this, I just quoted what Shreyas has mentioned in that PR thinking it would be better to have it same across providers.

Reply:
In the case of GCS, there is a restriction on operations performed on a regional endpoint, mentioned here - https://cloud.google.com/storage/docs/regional-endpoints#supported-operations. IIRC, we decided that copy operations could not be supported by etcdbr when a regional endpoint is used, because the buckets could be residing in a region outside the "restricted" region, and the copy would fail.

But you're right, etcd-backup-restore should not bother about this, and simply try to perform the operation the user has specified. If the copy fails due to the restriction, it's the user's duty to check and correct it. I will make this change when fixing https://github.com/gardener/etcd-backup-restore/issues/727